### PR TITLE
debugging: add steps to repro firelens locally

### PR DESCRIPTION
**Pull request by PettitWesley**
Signed-off-by: Wesley Pettit <wppttt@amazon.com>

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: troubleshooting/debugging.md [hunk 1]
```diff
@@ -593,6 +595,109 @@ To work around this issue, we suggest using the following option so that system
...
+				"type": "fluentbit",
+				"options": {
+					"config-file-type": "s3",
+					"config-file-value": "arn:aws:s3:::yourbucket/yourdirectory/extra.conf"
+				}
+			},
+```
+2. It uses logConfiguration to configure an output:
+```
+"logConfiguration": {
```
Comment (matthewfala) [inline troubleshooting/debugging.md:620]: Formatting issue

File: troubleshooting/debugging.md [hunk 2]
```diff
@@ -593,6 +595,109 @@ To work around this issue, we suggest using the following option so that system
...
+				 "options": {
+					"Name": "kinesis_firehose",
+					"region": "us-west-2",
+					"delivery_stream": "my-stream",
+					"retry_limit": "2"
+				}
+			}
+```
+
+Perform the follpwing steps.
```
Comment (matthewfala) [inline troubleshooting/debugging.md:631]: spelling error on "following"

File: troubleshooting/debugging.md [hunk 3]
```diff
@@ -593,6 +595,109 @@ To work around this issue, we suggest using the following option so that system
...
+1. Create a new directory to store the configuration for this repro attempt and `cd` into it. 
+2. Create a sub-directory called `socket`. This will store the unix socket that Fluent Bit recieves container stdout/stderr logs from. 
+3. Copy the custom config file into your current working directory. In this case, we will call it `extra.conf`.
+4. Create a file called `fluent-bit.conf` in yoru current working directory.
```
Comment (matthewfala) [inline troubleshooting/debugging.md:635]: spelling error on your

File: troubleshooting/debugging.md [hunk 4]
```diff
@@ -593,6 +595,109 @@ To work around this issue, we suggest using the following option so that system
...
+    d. Create an `[OUTPUT]` configuration based on the `logConfiguration` `options` map for each container in the task definition. The `options` simply become key value pairs in the `[OUTPUT]` configuration.
+    e. Your configuration is now ready to test! If you have a more complicated custom configuration file, you may need to perform additional steps like adding parser files to your working directly, and updating the import paths for them in the Fluent Bit `[SERVICE]` section if your custom config had that. 
+5. Next you need to run Fluent Bit. You can use the following command, which assumes you have an AWS credentials file in `$HOME/.aws` which you want to use for credentials. Run this command in the same working directory that you put your config files in. You may need to customize it to run a different image, mount a directory to dump core files (see our core file tutorial) or add environment variables.
```
Comment (matthewfala) [inline troubleshooting/debugging.md:641]: I personally would really benefit from a tree view of the directory structure: https://www.cyberciti.biz/faq/linux-show-directory-structure-command-line/

The instructions are clear, but I have a hard time visualizing what the resulting directory pattern would be.

File: troubleshooting/debugging.md [hunk 5]
```diff
@@ -593,6 +595,109 @@ To work around this issue, we suggest using the following option so that system
...
+[INPUT]
+    Name tcp
+    Tag firelens-healthcheck
+    Listen 127.0.0.1
+    Port 8877
+
+[FILTER]
+    Name   grep
+    Match app-firelens*
+    Regex  log [Ee]rror
```
Comment (matthewfala) [inline troubleshooting/debugging.md:678]: Does this only include logs with the text Error or error in the "log" field?

**Review by matthewfala — CHANGES_REQUESTED** [review]
Left a few comments. Only the spelling error ones should be fixed, the others are suggestions or clarification questions.

File: troubleshooting/debugging.md (hunk 5 above)
Comment (PettitWesley) [reply troubleshooting/debugging.md:678]: oops/thanks... I dunno why I made this mistake. I swear I remember when I was working on it that I said to myself, need to add a note about not copying just this one filter 🫠

File: troubleshooting/debugging.md (hunk 4 above)
Comment (PettitWesley) [reply troubleshooting/debugging.md:641]: Good idea. I think I will also edit this to be for the core-file option where you also mount a cores directory.

**Review by matthewfala — APPROVED** [review]

**Pull request merged**